espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/IntentHelpers.kt

```diff
@@ -243,6 +245,19 @@ fun makeIntent(context: Context?, options: Map<String, Any?>): Intent {
                             intent.putExtra(it.key as String, requireBool(it.key as String, it.value))
                         }
             },
+            "esa" to fun(key, value) {
+                requireMap(key, value)
+                        .filter { it.key is String }
+                        .forEach { entry ->
+                            requireString(entry.key as String, entry.value)
+                                    .split(",")
```

Comment:
this won't work if string items contain commas
Consider changing value type to an array

Reply:
I followed the adb sytax using the adb help. adb shell am

[--esal <EXTRA_KEY> <EXTRA_STRING_VALUE>[,<EXTRA_STRING_VALUE...]]
        (mutiple extras passed as List<String>; to embed a comma into a string,
         escape it using "\,")

But can change to a Map<String, List> instead of Map<String, String>. What do you recommend?

Reply:
in such case it would make sense to add proper handling of the escaped comma
Both ways would work for me